src/waveform/waveformwidgetfactory.cpp

```diff
@@ -1362,6 +1368,33 @@ void WaveformWidgetFactory::setUntilMarkTextHeightLimit(float value) {
     emit untilMarkTextHeightLimitChanged(value);
 }
 
+void WaveformWidgetFactory::setStemReorderOnChange(bool value) {
+    m_stemReorderOnChange = value;
+    if (m_config) {
+        m_config->setValue(ConfigKey("[Waveform]", "stem_reorder_on_change"),
+                value);
+    }
+    emit stemReorderOnChangeChanged(value);
+}
+
+void WaveformWidgetFactory::setStemOutlineOpacity(float value) {
+    m_stemOutlineOpacity = value;
+    if (m_config) {
+        m_config->setValue(ConfigKey("[Waveform]", "stem_outline_opacity"),
+                static_cast<double>(value));
+    }
+    emit stemOutlineOpacityChanged(value);
+}
+
+void WaveformWidgetFactory::setStemOpacity(float value) {
+    m_stemOpacity = value;
+    if (m_config) {
+        m_config->setValue(ConfigKey("[Waveform]", "stem_opacity"),
+                static_cast<double>(value));
+    }
+    emit stemOpacityChanged(value);
```

Comment:
NIT, no need to cast.
```suggestion
void WaveformWidgetFactory::setStemOutlineOpacity(float value) {
    m_stemOutlineOpacity = value;
    if (m_config) {
        m_config->setValue(ConfigKey("[Waveform]", "stem_outline_opacity"),
                value);
    }
    emit stemOutlineOpacityChanged(value);
}

void WaveformWidgetFactory::setStemOpacity(float value) {
    m_stemOpacity = value;
    if (m_config) {
        m_config->setValue(ConfigKey("[Waveform]", "stem_opacity"),
                value);
    }
    emit stemOpacityChanged(value);
```

Reply:
I'm sorry, this suggestion was BS... I've misread the code. Feel free to either revert or add the template specializations that make this work.